invenio_records_resources/references/entity_resolvers/results.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 Graz University of Technology.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Resolver for result items."""
+
+from invenio_access.permissions import system_identity
+
+from .base import EntityProxy, EntityResolver
+
+
+class ResultItemProxy(EntityProxy):
+    """Resolver proxy for a result item entity."""
+
+    def __init__(self, resolver, ref_dict, service):
+        """Constructor."""
+        super().__init__(resolver, ref_dict)
+        self.service = service
+
+    def _resolve(self):
+        """Resolve the result item from the proxy's reference dict."""
+        id_ = self._parse_ref_dict_id()
+        return self.service.read(system_identity, id_).to_dict()
+
+    def get_needs(self, ctx=None):
+        """Return needs."""
+        # TODO: service.config.permission_policy_cls.can_read or []
+        #       Since resolving is done by the system, might not be applicable
+        return []
+
+    def pick_resolved_fields(self, identity, resolved_dict):
+        """Select which fields to return when resolving the reference."""
+        return {"id": resolved_dict["id"]}
+
+
+class ResultItemResolver(EntityResolver):
+    """Resolver for result items."""
+
+    def __init__(self, item_cls, service_id, type_key, proxy_cls=ResultItemProxy):
+        """Constructor.
+
+        :param item_cls: The result item class to use.
+        :param service_id: The record service id.
+        :param type_key: The value to use for the TYPE part of the ref_dicts.
+        """
+        self.item_cls = item_cls
+        self.type_key = type_key
```

Comment:
Yes, good idea!
In most cases, the service id will be in plural (e.g. `communities`, `requests`). Since the same value is used for the ref dict and when resolving, it should be fine.

Reply:
Hm, now that I see it actually, the plural breaks a bit the intuitiveness of `{"request": "<req_id>"}` vs. `{"requests": "<req_id>"}`. Let's keep the `type_key`/`type_id` pattern, and get only the `item_cls` from the service config.